src/components/EmailResetForm.tsx

```diff
@@ -0,0 +1,156 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Button from "./Button-Link/Button/Button";
+import { faX } from "@fortawesome/free-solid-svg-icons";
+import { useState } from "react";
+
+const EmailResetForm = ({
+  initialValue,
+  onSave,
+  label,
+}: {
+  initialValue: string;
+  onSave: (email: string, password: string) => void;
+  label: string;
+}) => {
+  const [passwordValue, setPasswordValue] = useState("");
+  const [emailValue, setEmailValue] = useState(initialValue);
+  const [wizard, setWizard] = useState<
+    | {
+        type: "initial";
+      }
+    | {
+        type: "editable";
+      }
+    | {
+        type: "loading";
+      }
+    | {
+        type: "error";
+        message: string;
+      }
+  >({ type: "initial" });
+
+  const onSubmit: React.FormEventHandler<HTMLFormElement> = async (e) => {
+    e.preventDefault();
+    try {
+      setWizard({ type: "loading" });
+      await onSave(e.currentTarget.email.value, e.currentTarget.password.value);
+      setWizard({ type: "initial" });
+    } catch (err) {
+      setWizard({ type: "error", message: "Unable to change the email" });
```

Comment:
Same as the previous component, use the `message` from the `err`. That's because we first want the server message - if it exists - because it has the most context.

Only if we have no `message` on `err`, use the default one.